rust/operator-binary/src/backend/kerberos_keytab.rs

```diff
@@ -174,7 +174,9 @@ udp_preference_limit = 1
 
 [domain_realm]
 cluster.local = {realm_name}
+cluster.local. = {realm_name}
 .cluster.local = {realm_name}
+.cluster.local. = {realm_name}
```

Comment:
IME, this shouldn't be necessary at all (probably since we set the default realm before). But if we do keep it then we should read the actual cluster domain, not hard-code cluster.local specifically.

Reply:
Good catch, this PR was mostly about trying out if we can fix the TLS cert problems we have.
Fixed it in 5781e66